src/LinkDotNet.Blog.Web/Features/Bookmarks/BookmarkService.cs

```diff
@@ -0,0 +1,64 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using LinkDotNet.Blog.Domain;
+using LinkDotNet.Blog.Infrastructure.Persistence.Sql;
+using LinkDotNet.Blog.Web.Features.Services;
+using Microsoft.EntityFrameworkCore;
+
+namespace LinkDotNet.Blog.Web.Features.Bookmarks;
+
+public class BookmarkService : IBookmarkService
+{
+    private readonly ILocalStorageService localStorageService;
+
+    public BookmarkService(ILocalStorageService localStorageService)
+    {
+        this.localStorageService = localStorageService;
+    }
+
+    public async Task<bool> IsBookMarked(string postId)
+    {
+        if(string.IsNullOrEmpty(postId)) throw new ArgumentException(nameof(postId));
+        await InitializeIfNotExists();
+        var bookmarks = await localStorageService.GetItemAsync<HashSet<string>>("bookmarks");
+
+        return bookmarks.Contains(postId);
+    }
+
+    public async Task<IReadOnlyList<string>> GetBookmarkedPostIds()
+    {
+        await InitializeIfNotExists();
+        return await localStorageService.GetItemAsync<IReadOnlyList<string>>("bookmarks");
+    }
+
+    public async Task ToggleBookmark(string postId)
+    {
+        await InitializeIfNotExists();
+        if(string.IsNullOrEmpty(postId)) throw new ArgumentException(nameof(postId));
```

Comment:
The same here - plus: The sanity check should be the first line in this code. No need to call `InitializeIfNotExists` when postId is not set.